apps/server/src/challenges/challenges.controller.ts

```diff
@@ -0,0 +1,101 @@
+import {
+  ApiTags,
+  ApiBearerAuth,
+  ApiCreatedResponse,
+  ApiBadRequestResponse,
+} from '@nestjs/swagger';
+import {
+  Get,
+  Body,
+  Post,
+  Param,
+  Patch,
+  Delete,
+  HttpCode,
+  Controller,
+  HttpStatus,
+} from '@nestjs/common';
+
+import { Roles } from 'src/decorator/roles.decorator';
+
+import { RoleEnum } from '@/common/constants/role.enum';
+
+import {
+  CreateChallengeDto,
+  UpdateChallengeDto,
+} from '@/challenges/dto/challenges.dto';
+
+import { ChallengesService } from '@/challenges/challenges.service';
+
+import { Challenge } from '@/challenges/entities/challenges.entity';
+
+@ApiTags('Challenges')
+@Controller('challenges')
+@ApiBearerAuth()
+export class ChallengesController {
+  constructor(private readonly challengesService: ChallengesService) {}
+
+  @Post()
+  @Roles(RoleEnum.ADMIN)
+  @ApiCreatedResponse({
+    description: 'Created a new challenge',
+    type: Challenge,
+  })
+  @ApiBadRequestResponse({ description: 'Challenge cannot be created' })
+  @HttpCode(HttpStatus.CREATED)
```

Comment:
No need to specify `HttpCode`. Nest will resolve this for us.

Reply:
Hmm, I used this for swagger api. Is this not needed there as well?

Reply:
`@ApiOkResponse()` for 200 and `@ApiCreatedResponse` for 201

Reply:
okay